server/src/main/java/org/opensearch/identity/ApplicationAwareSubject.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.identity;
+
+import java.security.Principal;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import org.opensearch.cluster.ApplicationManager;
+import org.opensearch.identity.scopes.ApplicationScope;
+import org.opensearch.identity.scopes.Scope;
+import org.opensearch.identity.tokens.AuthToken;
+
+/**
+ * This class defines an ApplicationAwareSubject.
+ *
+ * An ApplicatioAwarenSubject is a type of Subject which wraps another basic Subject object.
+ * With an ApplicationAwareSubject, we represent a system that interacts with the Identity access control system.
+ *
+ * @opensearch.experimental
+ */
+@SuppressWarnings("overrides")
+public class ApplicationAwareSubject implements Subject {
```

Comment:
That should be `ExtenstionSubject` I think

Reply:
This is meant to represent an Application more generally. This will include Plugins and Extensions so ApplicationAwareSubject was picked. I can swap to ApplicationSubject if you prefer. The "Aware" comes from the fact that we wrap another existing subject. 

What do you prefer?

Reply:
I think it make sense, for some reasons it does not play well (to me) to classify Plugins and Extenstions as Applications (but I don't have any suitable alternative in my head to classify those under better name).

Reply:
It surely does, and no, I do not have better option to suggest (`module` does not fit either, those are plugins in OS world).

Reply:
I can live with PluginAwareSubject, does that seem better?

Reply:
> I can live with PluginAwareSubject, does that seem better?

Not really, in my opinion (application logically maps to plugin / module / extension)